contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -345,23 +384,79 @@ contract OETHVaultCore is VaultCore {
         if (_asset == weth) {
             WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
             // Need to remove WETH that is reserved for the withdrawal queue
-            return balance + queue.claimed - queue.queued;
+            if (balance + queue.claimed >= queue.queued) {
+                return balance + queue.claimed - queue.queued;
+            }
         }
     }
 
-    function _allocate() internal override {
+    /**
+     * @notice Allocate unallocated funds on Vault to strategies.
+     **/
+    function allocate() external override whenNotCapitalPaused nonReentrant {
         // Add any unallocated WETH to the withdrawal queue first
         _addWithdrawalQueueLiquidity();
 
-        super._allocate();
+        _allocate();
     }
 
+    /// @dev Allocate WETH to the default WETH strategy if there is excess to the Vault buffer.
+    /// This is called from either `mint` or `allocate` and assumes `_addWithdrawalQueueLiquidity`
+    /// has been called before this function.
+    function _allocate() internal override {
```

Comment:
Gotta say I love how linear this function is. ❤️

Reply:
it went through a number of iterations. Each time getting simpler